src/main/java/com/stripe/model/SetupAttempt.java

```diff
@@ -0,0 +1,247 @@
+  }
+
+  public void setPaymentMethodObject(PaymentMethod expandableObject) {
+    this.paymentMethod =
+        new ExpandableField<PaymentMethod>(expandableObject.getId(), expandableObject);
+  }
+
+  /** Get ID of expandable {@code setupIntent} object. */
+  public String getSetupIntent() {
+    return (this.setupIntent != null) ? this.setupIntent.getId() : null;
+  }
+
+  public void setSetupIntent(String id) {
+    this.setupIntent = ApiResource.setExpandableFieldId(id, this.setupIntent);
+  }
+
+  /** Get expanded {@code setupIntent}. */
+  public SetupIntent getSetupIntentObject() {
+    return (this.setupIntent != null) ? this.setupIntent.getExpanded() : null;
+  }
+
+  public void setSetupIntentObject(SetupIntent expandableObject) {
+    this.setupIntent = new ExpandableField<SetupIntent>(expandableObject.getId(), expandableObject);
+  }
+
+  /** Returns a list of SetupAttempts associated with a provided SetupIntent. */
+  public static SetupAttemptCollection list(Map<String, Object> params) throws StripeException {
+    return list(params, (RequestOptions) null);
+  }
+
+  /** Returns a list of SetupAttempts associated with a provided SetupIntent. */
+  public static SetupAttemptCollection list(Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    String url = String.format("%s%s", Stripe.getApiBase(), "/v1/setup_attempts");
+    return ApiResource.requestCollection(url, params, SetupAttemptCollection.class, options);
+  }
+
+  /** Returns a list of SetupAttempts associated with a provided SetupIntent. */
+  public static SetupAttemptCollection list(SetupAttemptListParams params) throws StripeException {
+    return list(params, (RequestOptions) null);
+  }
+
+  /** Returns a list of SetupAttempts associated with a provided SetupIntent. */
+  public static SetupAttemptCollection list(SetupAttemptListParams params, RequestOptions options)
+      throws StripeException {
+    String url = String.format("%s%s", Stripe.getApiBase(), "/v1/setup_attempts");
+    return ApiResource.requestCollection(url, params, SetupAttemptCollection.class, options);
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class PaymentMethodDetails extends StripeObject {
+    @SerializedName("card")
+    Card card;
+
+    /**
+     * The type of the payment method used in the SetupIntent (e.g., {@code card}). An additional
+     * hash is included on {@code payment_method_details} with a name matching this value. It
+     * contains confirmation-specific information for the payment method.
+     */
+    @SerializedName("type")
+    String type;
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class Card extends StripeObject {
+      /** Populated if this authorization used 3D Secure authentication. */
+      @SerializedName("three_d_secure")
+      Charge.PaymentMethodDetails.Card.ThreeDSecure threeDSecure;
```

Comment:
Flagging this should not be shared but for now Java does still share so deferring to a future major